config/oidc_backends_config.xml.sample

```diff
@@ -22,4 +22,20 @@
          login to Galaxy using their Google account.
         -->
     </provider>
+    <provider name="Keycloak-auth" implementation="Keycloak">
```

Comment:
I guess I was confused with "how keycloak knows that it should redirect the user to Google, for instance"; because I don't see any reference to IdP, so I assumed it is the `idp_hint` specifying this backend (as you also . put it in the commend). I guess it could be intuitive (and consistent with PSA config) if it reads as `<provider name="google" implementation="Keycloak">`.

If you leave it as `<provider name="Keycloak-auth" implementation="Keycloak">` and then you want to implement login with X and Y idps, then we might end-up in configs such as the following: 

```xml
    <provider name="Keycloak-auth" implementation="Keycloak">
        ...
        <idp_hint>X</idp_hint>
    </provider>
    <provider name="Keycloak-auth" implementation="Keycloak">
        ...
        <idp_hint>Y</idp_hint>
    </provider>
```
or: 
```xml
    <provider name="Keycloak-auth" implementation="Keycloak">
        ...
        <idps>
                <idp_hint>X</idp_hint>
                <idp_hint>Y</idp_hint>
        </idp_hint>
    </provider>
```
but I guess the following would be more intuitive: 
```xml
    <provider name="X" implementation="Keycloak">
        ...
    </provider>
    <provider name="Y" implementation="Keycloak">
        ...
    </provider>
```

Reply:
Yes, that last one is what I had in mind, you would create two provider entries that could have different idp_hints.